activerecord/lib/active_record/signed_id.rb

```diff
@@ -8,9 +8,20 @@ module SignedId
     included do
       ##
       # :singleton-method:
+      # DEPRECATION WARNING: This is deprecated and will be removed in Rails ?.?. The secret will be derived from 'secret_key_base' instead.
+      #
       # Set the secret used for the signed id verifier instance when using Active Record outside of \Rails.
       # Within \Rails, this is automatically set using the \Rails application key generator.
       class_attribute :signed_id_verifier_secret, instance_writer: false
+
+      # :nodoc:
+      class_attribute :signed_id_verifier, instance_writer: false
+
+      # This option allows for a smooth transition during the deprecation of the legacy <tt>signed_id_verifier</tt>. It supports the following modes:
+      # <tt>:generate_and_verify</tt> – Generates signed IDs using the legacy format while verifying both the new and legacy formats.
+      # <tt>:verify</tt> (default) – Generates signed IDs using the new format while verifying both the new and legacy formats.
+      # Note: The new format is not URL-safe unless <tt>Rails.application.message_verifiers</tt> is configured with <tt>url_safe: true</tt>.
+      mattr_accessor :use_legacy_signed_id_verifier, default: :verify
```

Comment:
❓ Should use `:generate_and_verify` as the default value until `message_verifiers` is updated to use `url_safe: true`?

Reply:
I tried to find the right place to define this config attribute. I also updated the `configuring.md`.